### ISSUE TITLE
`.nbNavigationDestination` computes its content multiple times on iOS 16+

### ISSUE THREAD

**Issue by jadar** [johnpatrickmorgan/NavigationBackport#47]
I ran into an issue today where `.nbNavigationDestination` would recompute the destination view, even when the path had not changed. Most of the time, this shouldn't be an issue since View's should be immutable. However, I ran into an issue with @ObservableObject and state resetting because of this. It also seems this is an issue because views should not need to be recomputed unless parent state changes, which isn't happening. This doesn't happen on the backport implementation, nor the vanilla Apple implementation. See the attached example project.

Click on an item in the list, click on "Toggle Sheet", and watch the sheet dismiss itself.

**Comment by johnpatrickmorgan** [comment]
Hi @jadar, thanks for raising this issue, sorry for the slow response. The zip file doesn't seem to include the `ContentView.swift` file, so I wasn't able to run it. Are you able to re-upload please?

**Comment by jadar** [comment]
@johnpatrickmorgan I apologize for my mistake. Not sure how I managed to do that. Here is an updated project with the requires files...
[NavigationBackportBug.zip](https://github.com/johnpatrickmorgan/NavigationBackport/files/13822316/NavigationBackportBug.zip)

**Comment by johnpatrickmorgan** [comment]
Thanks @jadar! This is similar to #46 . In my response there, I suggested that view models should be stored on the navigation stack to avoid new instances being created when a destination is recomputed. E.g. in your example you might resolve the issue by doing something like: 

```swift
NBNavigationStack(path: $path) {
    List {
        NBNavigationLink(value: ViewModel()) {
            Text("Item 1")
        }
    }
    .nbNavigationDestination(for: ViewModel.self) { vm in
        DetailView(viewModel: vm)
    }
}
``` 

However, I hadn't realised until you demonstarted it that the standard `NavigationStack`  behaviour was more permissive. And it's surprising that your issue only arises when `NBNavigationStack` is using a standard `NavigationStack` under the hood - the issue seems to be resolved if I add `.nbUseNavigationStack(.never)` to the `NBNavigationStack`. So it's worth investigating why this library's use of `NavigationStack` affects its behaviour.

**Comment by johnpatrickmorgan** [comment]
I made a [change](https://github.com/johnpatrickmorgan/NavigationBackport/commit/d1759195a9be9f182b105b1ccaceff573218af6f) so that the closure is only invoked once for the same data at the same index, shipped in `v0.9.4`. Thanks for your patience!
